.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,262 @@
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Verify Java installation
+        run: java -version
+
+      - uses: actions/cache@v4
+        name: Restore Gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+            rskj-core/build
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get Gradle jar
+        run: |
+          if [ ! -f ./gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+          fi
+
+      - name: Run SonarQube analysis
+        run: |
+          sudo apt-get update && sudo apt-get install -yqq git
+          extra_flags=""
+          if [ -n "$CIRCLE_PULL_REQUEST" ]; then
+            # https://community.sonarsource.com/t/no-code-or-issue-found-in-pull-request-decorations-github-circleci/8496
+            git branch -f master origin/master
+            # extract PR number, as PR URLs are in the form
+            # https://github.com/$user/$repo/pull/$pr_number
+            pr_number==${{ github.event.pull_request.number }}
+            extra_flags="-Dsonar.pullrequest.base=master
+              -Dsonar.pullrequest.branch=${{ github.head_ref }}
+              -Dsonar.pullrequest.key=$pr_number"
+          else
+            extra_flags="-Dsonar.branch.name=master"
+          fi
+          ./gradlew sonarqube --no-daemon -x build -x test \
+              $extra_flags \
+              -Dsonar.organization=rsksmart \
+              -Dsonar.host.url="https://sonarcloud.io" \
+              -Dsonar.login="${{ secrets.SONAR_TOKEN }}"
+
+  tests:
+    needs: build
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y curl
+      - name: Install Node.js
+        run: |
+          curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
+          apt-get install -y nodejs
+      - name: Check Node.js version
+        run: node --version
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get Gradle jar
+        run: |
+          if [ ! -f ./gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+          fi
+
+      - name: Unit tests
+        run: |
+          chmod +x ./gradlew
+          ./gradlew --no-daemon --stacktrace test
+      - name: Integration tests
+        run: |
+          chmod +x ./gradlew
```

Comment:
There is really no need for doing this in 2 steps.

```suggestion
```

Reply:
yea I guess not. not sure why I have it there. will remove